File: src/frontend/src/app/components/atendimento/tabela/model/tabela-dto.ts
import { Exclude, Expose, Type } from 'class-transformer';
import { TabelaItemDTO } from './tabela-item-dto';
import { TipoTabela } from './tipo-tabela.enum';

@Exclude()
export class TabelaDTO {
  @Expose() id?: string;
  @Expose() nome?: string;
  @Expose() tipo?: TipoTabela;
  @Expose() ativo?: boolean;

  @Expose()
  @Type(() => TabelaItemDTO)
  itens?: TabelaItemDTO[];

  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/tabela/dto/TabelaDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.tabela.dto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.tabela.TipoTabela;
import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class TabelaDTO implements DTO {

    private UUID id;
    private String nome;
    private TipoTabela tipo;
    private Boolean ativo;
    private List<TabelaItemDTO> itens;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
